### DISCUSSION TITLE
Nodeless edges

### DISCUSSION THREAD

**Discussion by josephclawrence** [xyflow/xyflow#3204] [New Features]
Perhaps there is already a way to do this but I don't think so. 

It would be helpful to be able to draw lines on the canvas. As a first step they could just be straight lines, with optional. Within the current framework I see these as nodeless edges, with markers and labels etc.

The use case is I guess generally speaking annotations to a diagram. Labelled indications of directions of flow in the diagram. Indications of flow or connection between two areas of a diagram, not necessarily between two specific groups of nodes, etc.

**Comment by bcakmakoglu** [comment]
Technically, there is a way to create edges that don't connect to any *visible* nodes.

Just create a tiny, invisible node for an edge and you'll have your desired effect.

[Here's an example.](https://codesandbox.io/s/busy-hooks-p8765k?file=/src/Flow.tsx)
